File: February 2026 GFG SOLUTION/February-25.py
class Solution:
    def longestSubarray(self, arr, k):
        mp = {}
        sum_val = ans = 0
        for i in range(len(arr)):
            sum_val += 1 if arr[i] > k else -1
            if sum_val > 0:
                ans = i + 1
            elif sum_val - 1 in mp:
                ans = max(ans, i - mp[sum_val - 1])
            if sum_val not in mp:
                mp[sum_val] = i
        return ans

File: February 2026 GFG SOLUTION/February-25.java
class Solution {
    public int longestSubarray(int[] arr, int k) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int sum = 0, ans = 0;
        for (int i = 0; i < arr.length; i++) {
            sum += (arr[i] > k) ? 1 : -1;
            if (sum > 0) ans = i + 1;
            else if (mp.containsKey(sum - 1)) ans = Math.max(ans, i - mp.get(sum - 1));
            if (!mp.containsKey(sum)) mp.put(sum, i);
        }
        return ans;
    }
}
